Release v1.4.0 (#2203)
nautobot/nautobot: nautobot/docs/plugins/development.md

    model = 'dcim.device'

    def detail_tabs(self):
        """
        You may define extra tabs to render on a model's detail page by utilizing this method.
        Each tab is defined as a dict in a list of dicts.

        For each of the tabs defined:
        - The <title> key's value will become the tab link's title.
        - The <url> key's value is used to render the HTML link for the tab

        These tabs will be visible (in this instance) on the Device model's detail page as
        set by the DeviceContent.model attribute "dcim.device"

        This example demonstrates defining two tabs. The tabs will be ordered by their position in list.
        """
        return [
            {
                "title": "Plugin Tab 1",
                "url": reverse("plugins:example_plugin:device_detail_tab_1", kwargs={"pk": self.context["object"].pk}),
            },
            {
                "title": "Plugin Tab 2",
                "url": reverse("plugins:example_plugin:device_detail_tab_2", kwargs={"pk": self.context["object"].pk}),
            },
        ]

template_extensions = [DeviceExtraTabs, SiteAnimalCount]
```

#### Adding Extra Tabs

> 

Reviewer: ```suggestion

_Added in version 1.4.0_ <!-- markdownlint-disable-line MD036 -->

```

        - models/
          - object_model.html  # "animal.html"
```

## Overriding Existing Functionality

### Replacing Views

> 

Reviewer: ```suggestion

_Added in version 1.4.0_ <!-- markdownlint-disable-line MD036 -->

```

You may override any of the core or plugin views by providing an `override_views` `dict` in a plugin's `views.py` file.

To override a view, you must specify the view's fully qualified name as the `dict` key which consists of the app name followed by the view's name separated by a colon, for instance `dcim:device`. The `dict` value should be the overriding view function.

A simple example to override the device detail view:

```python
# views.py
from django.shortcuts import HttpResponse
from nautobot.core.views import generic

class DeviceViewOverride(generic.View):
    def get(self, request, *args, **kwargs):
        return HttpResponse(("Hello world! I'm a view which "
                             "overrides the device object detail view."))

override_views = {
    "dcim:device": DeviceViewOverride.as_view(),
}
```

## Note URL Endpoint

> 
> New in Nautobot 1.4 is support for models that inherit from `PrimaryModel` and `OrganizationalModel` to have notes associated. In order to utilize this new feature you will need to add the endpoint to `urls.py`. Here is an option to be able to support 1.4 and older versions of Nautobot:

Reviewer: ```suggestion

_Added in version 1.4.0_ <!-- markdownlint-disable-line MD036 -->

Models that inherit from `PrimaryModel` and `OrganizationalModel` can have notes associated. In order to utilize this new feature you will need to add the endpoint to `urls.py`. Here is an option to be able to support both 1.4+ and older versions of Nautobot:
```
